### PR TITLE
roachtest: assign each test an owner

### PR DESCRIPTION
**Pull request by danhhz**
Previously, roachtest ownership was a combination of individual opinion and git blame. To make it easier on the release team as well as to support future automation, tag each test as it's registered with an owning team (not individual). To prevent new tests from being added without an owner, the roachtest registry will reject any tests that are missing it, which will fail the build.

Release note: None

### CODE REVIEW

**Comment by danhhz** [comment]
Output of `roachtest list`, which now includes the owner. A couple left to figure out before this merges.

```
acceptance/bank/cluster-recovery [kv]
acceptance/bank/node-restart [kv]
acceptance/bank/zerosum-splits [kv] (skipped: https://github.com/cockroachdb/cockroach/issues/33683 (runs into  various errors during its rebalances, see isExpectedRelocateError))
acceptance/build-analyze [kv]
acceptance/build-info [kv]
acceptance/cli/node-status [kv]
acceptance/cluster-init [kv]
acceptance/decommission [kv]
acceptance/event-log [kv]
acceptance/gossip/locality-address [kv]
acceptance/gossip/peerings [kv]
acceptance/gossip/restart [kv]
acceptance/gossip/restart-node-one [kv]
acceptance/rapid-restart [kv]
acceptance/status-server [kv]
acceptance/version-upgrade [kv] (skipped: #43957)
alterpk [sql-exec]
backup2TB/n10cpu4 [bulkio]
backupTPCC [bulkio]
cancel/tpcc/distsql/w=10,nodes=3 [WIP]
cancel/tpcc/local/w=10,nodes=3 [WIP]
cdc/bank [cdc]
cdc/cloud-sink-gcs/rangefeed=true [cdc]
cdc/crdb-chaos/rangefeed=true [cdc] (skipped: #37716)
cdc/initial-scan/rangefeed=true [cdc]
cdc/ledger/rangefeed=true [cdc]
cdc/poller/rangefeed=false [cdc]
cdc/schemareg [cdc]
cdc/sink-chaos/rangefeed=true [cdc]
cdc/tpcc-1000/rangefeed=true [cdc]
clearrange/checks=false [kv]
clearrange/checks=true [kv]
clock/jump/large_backward_disabled [kv]
clock/jump/large_backward_enabled [kv]
clock/jump/large_forward_disabled [kv]
clock/jump/large_forward_enabled [kv]
clock/jump/small_forward_enabled [kv]
clock/monotonic/persistent [kv]
copy/bank/rows=10000000,nodes=9,txn=false [WIP]
copy/bank/rows=10000000,nodes=9,txn=true [WIP]
decommission/nodes=4/duration=1h0m0s [kv]
disk-full [kv] (skipped: https://github.com/cockroachdb/cockroach/issues/35328#issuecomment-478540195)
disk-stalled/log=false,data=false [kv]
disk-stalled/log=false,data=true [kv]
disk-stalled/log=true,data=false [kv]
disk-stalled/log=true,data=true [kv]
django [appdev] (skipped: django tests are still too flaky to run)
drop/tpcc/w=100,nodes=9 [WIP]
election-after-restart [kv] (skipped: https://github.com/cockroachdb/cockroach/issues/35047)
encryption/nodes=1 [WIP]
flowable [WIP]
follower-reads/nodes=3 [kv]
gopg [appdev]
gossip/chaos/nodes=9 [kv]
hibernate [appdev]
hotspotsplits/nodes=4 [kv]
import/tpcc/warehouses=1000/nodes=32 [bulkio]
import/tpcc/warehouses=1000/nodes=4 [bulkio]
import/tpcc/warehouses=4000/geo [bulkio]
import/tpch/nodes=8 [bulkio]
inconsistency [kv]
indexes/2/nodes=6/multi-region [kv]
interleavedpartitioned [WIP]
jepsen/bank-multitable/majority-ring [kv]
jepsen/bank-multitable/majority-ring-start-kill-2 [kv]
jepsen/bank-multitable/parts-start-kill-2 [kv]
jepsen/bank-multitable/split [kv]
jepsen/bank-multitable/start-kill-2 [kv]
jepsen/bank-multitable/start-stop-2 [kv]
jepsen/bank-multitable/strobe-skews [kv]
jepsen/bank/majority-ring [kv]
jepsen/bank/majority-ring-start-kill-2 [kv]
jepsen/bank/parts-start-kill-2 [kv]
jepsen/bank/split [kv]
jepsen/bank/start-kill-2 [kv]
jepsen/bank/start-stop-2 [kv]
jepsen/bank/strobe-skews [kv]
jepsen/g2/majority-ring [kv]
jepsen/g2/majority-ring-start-kill-2 [kv]
jepsen/g2/parts-start-kill-2 [kv]
jepsen/g2/split [kv]
jepsen/g2/start-kill-2 [kv]
jepsen/g2/start-stop-2 [kv]
jepsen/g2/strobe-skews [kv]
jepsen/monotonic/majority-ring [kv]
jepsen/monotonic/majority-ring-start-kill-2 [kv]
jepsen/monotonic/parts-start-kill-2 [kv]
jepsen/monotonic/split [kv]
jepsen/monotonic/start-kill-2 [kv]
jepsen/monotonic/start-stop-2 [kv]
jepsen/monotonic/strobe-skews [kv]
jepsen/multi-register/majority-ring [kv]
jepsen/multi-register/majority-ring-start-kill-2 [kv]
jepsen/multi-register/parts-start-kill-2 [kv]
jepsen/multi-register/split [kv]
jepsen/multi-register/start-kill-2 [kv]
jepsen/multi-register/start-stop-2 [kv]
jepsen/multi-register/strobe-skews [kv]
jepsen/register/majority-ring [kv]
jepsen/register/majority-ring-start-kill-2 [kv]
jepsen/register/parts-start-kill-2 [kv]
jepsen/register/split [kv]
jepsen/register/start-kill-2 [kv]
jepsen/register/start-stop-2 [kv]
jepsen/register/strobe-skews [kv]
jepsen/sequential/majority-ring [kv]
jepsen/sequential/majority-ring-start-kill-2 [kv]
jepsen/sequential/parts-start-kill-2 [kv]
jepsen/sequential/split [kv]
jepsen/sequential/start-kill-2 [kv]
jepsen/sequential/start-stop-2 [kv]
jepsen/sequential/strobe-skews [kv]
jepsen/sets/majority-ring [kv]
jepsen/sets/majority-ring-start-kill-2 [kv]
jepsen/sets/parts-start-kill-2 [kv]
jepsen/sets/split [kv]
jepsen/sets/start-kill-2 [kv]
jepsen/sets/start-stop-2 [kv]
jepsen/sets/strobe-skews [kv]
kv/contention/nodes=4 [kv]
kv/gracefuldraining/nodes=3 [kv] (skipped: https://github.com/cockroachdb/cockroach/issues/33501)
kv/quiescence/nodes=3 [kv]
kv/splits/nodes=3/quiesce=false [kv]
kv/splits/nodes=3/quiesce=true [kv]
kv0/enc=false/nodes=1 [kv]
kv0/enc=false/nodes=1/cpu=32 [kv]
kv0/enc=false/nodes=3 [kv]
kv0/enc=false/nodes=3/batch=16 [kv]
kv0/enc=false/nodes=3/cpu=32 [kv]
kv0/enc=false/nodes=3/cpu=32/seq [kv]
kv0/enc=false/nodes=3/cpu=32/size=4kb [kv]
kv0/enc=false/nodes=3/cpu=32/size=64kb [kv]
kv0/enc=false/nodes=3/cpu=32/splt=0 [kv]
kv0/enc=false/nodes=3/cpu=96 [kv]
kv0/enc=false/nodes=3/size=4kb [kv]
kv0/enc=false/nodes=3/size=64kb [kv]
kv0/enc=false/nodes=3/splt=0 [kv]
kv0/enc=true/nodes=1 [kv]
kv0/enc=true/nodes=3 [kv]
kv0/weekly/enc=false/nodes=32 [kv] (skipped: [tag:default] does not match [weekly])
kv0bench/nodes=10/cpu=8/sequential [kv]
kv0bench/nodes=10/cpu=8/shards=20/sequential [kv]
kv0bench/nodes=20/cpu=8/sequential [kv]
kv0bench/nodes=20/cpu=8/shards=80/sequential [kv]
kv0bench/nodes=5/cpu=8/sequential [kv]
kv0bench/nodes=5/cpu=8/sequential/2nd_idx [kv]
kv0bench/nodes=5/cpu=8/shards=10/sequential [kv]
kv0bench/nodes=5/cpu=8/shards=10/sequential/2nd_idx [kv]
kv50/enc=false/nodes=4/cpu=96/batch=64 [kv]
kv50/rangelookups/relocate/nodes=8 [kv]
kv50/rangelookups/split/nodes=8 [kv]
kv95/enc=false/nodes=1 [kv]
kv95/enc=false/nodes=1/cpu=32 [kv]
kv95/enc=false/nodes=3 [kv]
kv95/enc=false/nodes=3/batch=16 [kv]
kv95/enc=false/nodes=3/cpu=32 [kv]
kv95/enc=false/nodes=3/cpu=32/seq [kv]
kv95/enc=false/nodes=3/cpu=32/size=4kb [kv]
kv95/enc=false/nodes=3/cpu=32/size=64kb [kv]
kv95/enc=false/nodes=3/cpu=32/splt=0 [kv]
kv95/enc=false/nodes=3/cpu=96 [kv]
kv95/enc=false/nodes=3/size=4kb [kv]
kv95/enc=false/nodes=3/size=64kb [kv]
kv95/enc=false/nodes=3/splt=0 [kv]
kv95/enc=true/nodes=1 [kv]
kv95/enc=true/nodes=3 [kv]
kv95/weekly/enc=false/nodes=32 [kv] (skipped: [tag:default] does not match [weekly])
ledger/nodes=6/multi-az [kv]
lib/pq [appdev]
network/sanity/nodes=4 [kv]
network/tpcc/nodes=4 [kv]
overload/tpcc_olap/nodes=3/cpu=8/w=50/c=96 [kv]
pgjdbc [appdev]
pgx [appdev]
psycopg [appdev]
queue/nodes=1 [kv] (skipped: https://github.com/cockroachdb/cockroach/issues/17229)
rebalance-leases-by-load [kv]
rebalance-replicas-by-load [kv]
replicagc-changed-peers/noRestart [kv]
replicagc-changed-peers/withRestart [kv]
replicate/rebalance/3to5 [kv]
replicate/up/1to3 [kv]
replicate/wide [kv]
restart/down-for-2m [kv]
restore2TB/nodes=10 [bulkio]
restore2TB/nodes=32 [bulkio]
roachmart/partition=false [WIP]
roachmart/partition=true [WIP]
scaledata/distributed_semaphore/nodes=3 [kv]
scaledata/distributed_semaphore/nodes=6 [kv]
scaledata/filesystem_simulator/nodes=3 [kv]
scaledata/filesystem_simulator/nodes=6 [kv]
scaledata/jobcoordinator/nodes=3 [kv]
scaledata/jobcoordinator/nodes=6 [kv]
schemachange/bulkingest [WIP]
schemachange/index/tpcc/w=100 [WIP]
schemachange/index/tpcc/w=1000 [WIP]
schemachange/invertedindex [WIP]
schemachange/mixed/kv [WIP]
schemachange/mixed/tpcc [WIP]
scrub/all-checks/tpcc/w=100 [WIP]
scrub/index-only/tpcc/w=100 [WIP]
secondary-index-multi-version [sql-exec]
splits/largerange/size=10GiB,nodes=3 [kv]
splits/load/sequential/nodes=3 [kv]
splits/load/spanning/nodes=3 [kv]
splits/load/uniform/nodes=3 [kv]
sqlalchemy [appdev]
sqlsmith/setup=empty/setting=default [sql-exec]
sqlsmith/setup=empty/setting=no-ddl [sql-exec]
sqlsmith/setup=empty/setting=no-mutations [sql-exec]
sqlsmith/setup=rand-tables/setting=default [sql-exec]
sqlsmith/setup=rand-tables/setting=no-ddl [sql-exec]
sqlsmith/setup=rand-tables/setting=no-mutations [sql-exec]
sqlsmith/setup=seed-vec/setting=vec [sql-exec]
sqlsmith/setup=seed/setting=default [sql-exec]
sqlsmith/setup=seed/setting=no-ddl [sql-exec]
sqlsmith/setup=seed/setting=no-mutations [sql-exec]
sqlsmith/setup=tpcc/setting=ddl-nodrop [sql-exec]
sqlsmith/setup=tpcc/setting=default [sql-exec]
sqlsmith/setup=tpcc/setting=no-ddl [sql-exec]
sqlsmith/setup=tpcc/setting=no-mutations [sql-exec]
sqlsmith/setup=tpch-sf1/setting=default [sql-exec]
sqlsmith/setup=tpch-sf1/setting=no-ddl [sql-exec]
sqlsmith/setup=tpch-sf1/setting=no-mutations [sql-exec]
synctest [storage]
sysbench/oltp_delete/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_insert/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_point_select/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_read_only/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_read_write/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_update_index/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_update_non_index/nodes=3/cpu=32/conc=128 [kv]
sysbench/oltp_write_only/nodes=3/cpu=32/conc=128 [kv]
tpcc-nowait/nodes=3/w=1 [kv]
tpcc/headroom/n4cpu16 [kv]
tpcc/mixed-headroom/n5cpu16 [kv]
tpcc/w=100/nodes=3/chaos=true [kv]
tpccbench/nodes=12/cpu=16 [kv] (skipped: [tag:default] does not match [weekly])
tpccbench/nodes=3/cpu=16 [kv]
tpccbench/nodes=3/cpu=4 [kv]
tpccbench/nodes=6/cpu=16/multi-az [kv]
tpccbench/nodes=9/cpu=4/chaos/partition [kv]
tpccbench/nodes=9/cpu=4/multi-region [kv]
tpchbench/sql20/nodes=3/cpu=4/sf=1 [sql-exec]
tpchbench/tpch/nodes=3/cpu=4/sf=1 [sql-exec]
tpchvec [sql-exec]
typeorm [appdev]
upgrade [kv]
version/mixed/nodes=3 [kv]
version/mixed/nodes=5 [kv]
weekly/tpcc-max [kv] (skipped: [tag:default] does not match [weekly])
ycsb/A/nodes=3 [kv]
ycsb/A/nodes=3/cpu=32 [kv]
ycsb/B/nodes=3 [kv]
ycsb/B/nodes=3/cpu=32 [kv]
ycsb/C/nodes=3 [kv]
ycsb/C/nodes=3/cpu=32 [kv]
ycsb/D/nodes=3 [kv]
ycsb/D/nodes=3/cpu=32 [kv]
ycsb/E/nodes=3 [kv]
ycsb/E/nodes=3/cpu=32 [kv]
ycsb/F/nodes=3 [kv]
ycsb/F/nodes=3/cpu=32 [kv]
```

**Review by petermattis — APPROVED** [review]
:lgtm: though I didn't review all of the assignments in detail. Glad to see this get done. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-:-LzNlEPe0oclIZ5FVC-l:bxv7ife)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @danhhz, @dt, and @tbg)

___
*[pkg/cmd/roachtest/clearrange.go, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-LzNkej62anMbQbeO4ZI:-LzNkej62anMbQbeO4ZJ:bn6bvsz) ([raw file](https://github.com/cockroachdb/cockroach/blob/df6a1f077cb46ed0ec2fdf6dfec7b565052ad994/pkg/cmd/roachtest/clearrange.go#L26)):*
> ```Go
> 		r.Add(testSpec{
> 			Name:  fmt.Sprintf(`clearrange/checks=%t`, checks),
> 			Owner: `kv`,
> ```

This could be `storage`. The majority of problems here have been due to RocksDB range tombstones.
___
*[pkg/cmd/roachtest/test_registry.go, line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-LzNkucD42YL5ZLD6-qy:-LzNkucD42YL5ZLD6-qz:b-6vefc2) ([raw file](https://github.com/cockroachdb/cockroach/blob/df6a1f077cb46ed0ec2fdf6dfec7b565052ad994/pkg/cmd/roachtest/test_registry.go#L38)):*
> ```Go
> 	`sql-exec`: struct{}{},
> 	`storage`:  struct{}{},
> 	`WIP`:      struct{}{},
> ```

Any reason you didn't define an `Owner` type and a list of constants? This could then be a map from the constant to the name of the team. Doesn't matter much. This is probably fine.

File: pkg/cmd/roachtest/test_registry.go
Comment (jlinder) [inline pkg/cmd/roachtest/test_registry.go]: Maybe make the keys a specific type / enum?

**Review by jlinder — APPROVED** [review]
LGTM. 

Left one thought.

**Review by danhhz — COMMENTED** [review]
I diffed this with the doc to make sure I didn't have any copy paste errors (it was fine). In the interest of moving this forward, I made the following guesses for the stragglers. We can always edit them later if they're wrong.

```
cancel/tpcc/distsql/w=10,nodes=3 [sql-exec]
cancel/tpcc/local/w=10,nodes=3 [sql-exec]
copy/bank/rows=10000000,nodes=9,txn=false [kv]
copy/bank/rows=10000000,nodes=9,txn=true [kv]
drop/tpcc/w=100,nodes=9 [kv]
encryption/nodes=1 [storage]
flowable [appdev]
interleavedpartitioned [partitioning]
roachmart/partition=false [partitioning]
roachmart/partition=true [partitioning]
schemachange/bulkingest [schema-change]
schemachange/index/tpcc/w=100 [schema-change]
schemachange/index/tpcc/w=1000 [schema-change]
schemachange/invertedindex [schema-change]
schemachange/mixed/kv [schema-change]
schemachange/mixed/tpcc [schema-change]
scrub/all-checks/tpcc/w=100 [sql-exec]
scrub/index-only/tpcc/w=100 [sql-exec]
```

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-:-LzNzL5v3xSKmFUMBPeV:b-xck983)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @jlinder, @petermattis, and @tbg)

___
*[pkg/cmd/roachtest/clearrange.go, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-LzNkej62anMbQbeO4ZI:-LzNnHPT6SEYvkoM63bW:bs7wumx) ([raw file](https://github.com/cockroachdb/cockroach/blob/df6a1f077cb46ed0ec2fdf6dfec7b565052ad994/pkg/cmd/roachtest/clearrange.go#L26)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

This could be `storage`. The majority of problems here have been due to RocksDB range tombstones.
</blockquote></details>

Done. 
___
*[pkg/cmd/roachtest/test_registry.go, line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-LzNkucD42YL5ZLD6-qy:-LzNmfuI3mqbHE4Xdav4:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/df6a1f077cb46ed0ec2fdf6dfec7b565052ad994/pkg/cmd/roachtest/test_registry.go#L38)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Any reason you didn't define an `Owner` type and a list of constants? This could then be a map from the constant to the name of the team. Doesn't matter much. This is probably fine.
</blockquote></details>

Done
___
*[pkg/cmd/roachtest/test_registry.go, line 39 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-LzNdU8PZA7M8TbnxMt_-r1-39:-LzNyR-1BGwZGVjmiTM-:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/df6a1f077cb46ed0ec2fdf6dfec7b565052ad994/pkg/cmd/roachtest/test_registry.go#L39)):*
<details><summary><i>Previously, jlinder (James H. Linder) wrote…</i></summary><blockquote>

Maybe make the keys a specific type / enum?
</blockquote></details>

Done

**Review by petermattis — APPROVED** [review]
> We can always edit them later if they're wrong.

Agreed. Are you planning to follow-up with a PR addressing the TODO?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44351#-:-LzO-AeL4oauHZPqVhNP:b-hn8esn)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @jlinder, and @tbg)

**Review by dt — APPROVED** [review]
🎉 

nit: s/schema-change/sql-schema/ (the sql-schema team's responsibility extends beyond just changes).

I don't know if it makes sense to insist that the "owner" is also a valid slack channel or mailing list or gh team, to additionally make it easier to (automatically?) surface failures to the right people, but we can so a find/replace later if we decide to do that, or make the owners map struct values actually hold more owner metadata or something. I also wonder if it'd make sense to start prefixing the owner to the name -- some names already do this -- but it'd help logically group related tests. In both cases these are future improvements that shouldn't hold up anything here.

**Comment by danhhz** [comment]
Thanks for the reviews! Will bors on green

Slack room is exactly the sort of metadata I was thinking of, so went ahead and added an OwnerMetadata struct, filling that one in. The code doesn't use it yet, but it will already be useful for the release engineer. Going to leave extending it with github tags/projects whatever for a followup since I'm not sure I know yet what we're going to do there.

**Comment by jlinder** [comment]
> I also wonder if it'd make sense to start prefixing the owner to the name -- some names already do this -- but it'd help logically group related tests. 

If we did this and ownership changed (like if a team became big enough that it split responsibilities between multiple teams), how would changing the names of the tests affect auto-generated issues and tracking issues through the name changes?

**Comment by danhhz** [comment]
bors r=petermattis,jlinder,dt

**Comment by otan** [comment]
Did we also want to tag ones that shouldn't block a release?

**Comment by danhhz** [comment]
I would rather we stop running them as part of the release process

**Comment by dt** [comment]
My understanding was that we wanted to do both of those things -- we'd stop running them manually as part of the release _process_ but they are still an important part of release _qualification_, but we can use the fact that they are already being run on every branch we release rather than re-running them during the release process (by just treating exiting failures as blockers). If a particular test _isn't_ a useful release qualifier though, it could choose to opt-out of its failures being treated (labeled) as blockers (e.g. I believe the sql team says that regressions caught by sqlsmith are not release blockers).

**Comment by danhhz** [comment]
Right. My "I would rather we stop running them as part of the release process" is only meant to be for sqlsmith. We've been told those are never release blockers and they are not investigated when the fail. So for those ones, I'd rather we didn't run them. Maybe we can use roachtest tags for this?

**Comment by dt** [comment]
In any case, seems like a separate enhancement, so let's update / discuss on the tracking meta-issue over at #43920 now that this is merged.

**Pull request merged**
